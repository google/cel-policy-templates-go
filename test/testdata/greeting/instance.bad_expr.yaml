# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

apiVersion: policy.acme.co/v1
kind: greeting
metadata:
  name: errant-greetings
selector:
  matchLabels:
  matchExpressions:
    - {key: "env", operator: "NotIn", values: ["test", "staging"]}
    - {key: "env", operator: "In", values: [["test"]]}
    - {key: "trace", operator: "DoesNotExists"}
rules:
  - greeting: "Goodbye"
  - farewell: "Hello"
  - greeting: "Happy New Year's!"
    conditions:
      - description: Ring in the New Year.
        expression: >
          request.time.getMonth() == 0 &&
          request.time.getDate() == 1"
