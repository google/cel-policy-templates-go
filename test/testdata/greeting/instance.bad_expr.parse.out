1~# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

2~apiVersion: 3~"policy.acme.co/v1"
4~kind: 5~"greeting"
6~metadata:7~
  8~name: 9~"errant-greetings"
10~selector:11~
  12~matchLabels: 13~
  14~matchExpressions:15~
    - 16~{17~key: 18~"env", 19~operator: 20~"NotIn", 21~values: 22~[23~"test", 24~"staging"]}
    - 25~{26~key: 27~"env", 28~operator: 29~"In", 30~values: 31~[32~[33~"test"]]}
    - 34~{35~key: 36~"trace", 37~operator: 38~"DoesNotExists"}
39~rules:40~
  - 41~42~greeting: 43~"Goodbye"
  - 44~45~farewell: 46~"Hello"
  - 47~48~greeting: 49~"Happy New Year's!"
    50~conditions:51~
      - 52~53~description: 54~"Ring in the New Year."
        55~expression: 56~>
          request.time.getMonth() == 0 &&
          request.time.getDate() == 1"
